### ISSUE TITLE
Unable to reproduce WTK results on some datasets

### ISSUE THREAD

**Issue by abalhomaid** [BorgwardtLab/WTK#1]
Hello,

Thank you for providing a resourceful package with documentation. 

I was wondering how to reproduce the results shown in results.csv. For a minimal example, I tried running WTK on 'ArrowHead' dataset with k=10. Below is the sample code

```
from wtk import transform_to_dist_matrix
from wtk.utilities import get_ucr_dataset, krein_svm_grid_search

# Read UCR data
X_train, y_train, X_test, y_test = get_ucr_dataset('./data/UCR/raw_data/', 'ArrowHead')

# Compute wasserstein distance matrices with subsequence length k=10
D_train, D_test = transform_to_dist_matrix(X_train, X_test, 10)

# Run the grid search
svm_clf = krein_svm_grid_search(D_train, D_test, y_train, y_test)

y_pred = svm_clf.predict(D_test)
accuracy_score(y_pred, y_test)
```
The output was

```
Best C: 100000.0
Best gamma: 0.0001

0.16
```
While in results.csv, the accuracy for 'ArrowHead' is 0.805714286. Could you please help?

Thank you for your time.

**Comment by chrisby** [comment]
Dear @abalhomaid 

thanks for opening the issue. I took a look at the `ArrowHead` data set and I can reproduce our results. Unfortunately, the issue you are facing stems from some unmentioned and potentially confusing implementation details: 

When you run `krein_svm_grid_search`, we perform CV on different `SVM` parameter **and** different kernel weight factors (lambda in the paper). Once the best parameters were found **on** the training set the respective kernel matrix is computed on the test set. On this matrix, we evaluate the SVM. Your code uses the distance matrix, not the correct kernel matrix when doing the predictions. 

Furthermore, as we described in the paper, we scout for 3 different subsequence lengths for each data set. For `ArrowHead`, the best `k` is `30%` which equates to `k=75`.  If you use `k=75` and set the debug level to debug, the test performance is actually written out and you should see the promised performance.
 ```
import logging
logging.basicConfig(level=logging.DEBUG)
```

I admit the behaviour of `krein_svm_grid_search` is not so intuitive. I will slightly change the API such that it also returns the correct test **Gram** matrix.

**Comment by chrisby** [comment]
In fact, you should be able to generate the kernel matrix from the output (gamma) you get:

```
K_test = get_kernel_matrix(D_test, psd=False, gamma=0.0001)
y_pred = svm_clf.predict(K_test)
accuracy_score(y_pred, y_test)
```

**Comment by abalhomaid** [comment]
Dear @chrisby 

Thank you for your quick response. I am able to reproduce the result on `ArrowHead` dataset now.
